src/library_webgpu.js

```diff
@@ -1690,14 +1807,29 @@ var LibraryWebGPU = {
     pipeline.label = UTF8ToString(labelPtr);
   },
 
+  // wgpuBindGroup
+
+  wgpuBindGroupSetLabel: function(bindGroupId, labelPtr) {
+    var bindGroup = WebGPU.mgrBindGroup.get(bindGroupId);
+    bindGroup.label = UTF8ToString(labelPtr);
+  },
+
+  // wgpuBindGroupLayout
+
+  wgpuBindGroupLayoutSetLabel: function(bindGroupLayoutId, labelPtr) {
+    var bindGroupLayout = WebGPU.mgrBindGroupLayout.get(bindGroupLayoutId);
+    bindGroupLayout.label = UTF8ToString(labelPtr);
+  },
+
   // wgpuBuffer
 
+  // In webgpu.h offset and size are passed in as size_t.
+  // And we assume that size_t is always 32bit in emscripten.
   wgpuBufferGetConstMappedRange: function(bufferId, offset, size) {
     var bufferWrapper = WebGPU.mgrBuffer.objects[bufferId];
     {{{ gpu.makeCheckDefined('bufferWrapper') }}}
 
-    // TODO: if the sentinel value becomes WGPU_WHOLE_SIZE instead of 0, update this.
-    if (size === 0) size = undefined;
+    if ((size >>> 0) === {{{ gpu.WHOLE_MAP_SIZE }}}) size = undefined;
```

Comment:
We also want the unsigned version to get passed to the `getMappedRange` call later, so probably should do
```suggestion
    size = size >>> 0;
    if (size === {{{ gpu.WHOLE_MAP_SIZE }}}) size = undefined;
```